lib/recurly/schemas.js

```diff
@@ -0,0 +1,156 @@
+const inspect = require('util').inspect
+const utils = require('./utils')
+
+/**
+ * Represents a primitive:
+ *   - String
+ *   - Number
+ *   - Boolean
+ *   - Object (plain js object)
+ *   - Array (of primitives only)
+ */
+class PrimitiveProperty {
+  isValid (value) {
+    return true
+  }
+
+  cast (value) {
+    return value
+  }
+}
+
+/**
+ * Represents a Date in the schema
+ */
+class DateProperty {
+  isValid (value) {
+    return typeof value === 'string'
+  }
+  cast (value) {
+    return new Date(value)
+  }
+}
+
+/**
+ * Represents an embedded Resource in the schema
+ */
+class ResourceProperty {
+  constructor (type) {
+    this.type = type
+  }
+
+  isValid (value) {
+    if (Array.isArray(value)) return false
+    return typeof value === 'object'
```

Comment:
I don't know if this will end up being relevant, but I think it's necessary to call out that if `value === null` this function will return true

Reply:
There is a null check before all of these functions are called.